examples/vae/net.py

```diff
@@ -1,67 +1,103 @@
-import six
+import numpy as np
 
 import chainer
+from chainer import cuda
+import chainer.distributions as D
 import chainer.functions as F
-from chainer.functions.loss.vae import gaussian_kl_divergence
 import chainer.links as L
+from chainer import reporter
 
 
-class VAE(chainer.Chain):
-    """Variational AutoEncoder"""
+class AvgELBOLoss(chainer.Chain):
+    def __init__(self, encoder, decoder, prior, beta=1.0, k=1):
+        super(AvgELBOLoss, self).__init__()
+        self.beta = beta
+        self.k = k
+
+        with self.init_scope():
+            self.encoder = encoder
+            self.decoder = decoder
+            self.prior = prior
+
+        self.loss = None
+        self.rec = None
+        self.penalty = None
```

Comment:
It seems these loss values don't need to be saved to attributes of the chain.  How about `reporter.report({'loss':, loss}, self)` etc?

Reply:
fixed them: d20a9b9